Add verifyToken() to Backend SDK docs (#1184)
clerk/clerk-docs: docs/references/backend/verify-token.mdx

---
title: verifyToken()
description: Use Clerk's Backend SDK to verify a token signature.
---

{/* JS page: https://github.com/clerk/javascript/blob/843e96e505cd91d676f0736e24465726b55014a7/packages/backend/src/tokens/verify.ts#L13 */}

> # `verifyToken()`

Reviewer: I feel like we might want to call out that this is a lower-level method and intended for more advanced use-cases, in most cases we want to recommend that people use `authenticateRequest()` instead if they are securing a more traditional API.

> Verifies a token signature.

Reviewer: ```suggestion
Verifies a Clerk-generated token signature.
```

```js
function verifyToken: (token: string, options: VerifyTokenOptions) => Promise<JwtReturnType<JwtPayload, TokenVerificationError>>;
```

## Parameters

| Name | Type | Description |
| --- | --- | --- |
| `token` | `string` | The token to verify. |
| `options` | `VerifyTokenOptions` | Options for verifying the token. |

### `VerifyTokenOptions`

> [!WARNING]
> You must provide either `jwtKey` or `secretKey`.

| Name | Type | Description |
| --- | --- | --- |
| `audience?` | `string \| string[]` | A string or list of [audiences](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3). |
| `authorizedParties?`| `string[]` | | A list of authorized parties. For example, `['http://localhost:4003', 'https://clerk.com']` |
| `clockSkewInMs?` | `number` | Specifies the allowed time difference (in milliseconds) between the Clerk server (which generates the token) and the clock of the user's application server when validating a token. Defaults to 5000 ms (5 seconds).|

> | `jwtKey?` | `string` | The JWT Public Key from the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys) page -> Show JWT public key -> JWKS URL** section of the Clerk Dashboard. It's recommended to use [the environment variable](/docs/deployments/clerk-environment-variables) instead. |

Reviewer: ```suggestion
| `jwtKey?` | `string` | The JWT Public Key from the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys) page -> Show JWT public key -> PEM Public Key** section of the Clerk Dashboard. It's recommended to use [the environment variable](/docs/deployments/clerk-environment-variables) instead. |
```

| `secretKey?` | `string` | The Clerk secret key from the [API Keys](https://dashboard.clerk.com/last-active?path=api-keys) page in the Clerk Dashboard.  |

> | `skipJwksCache?` | `boolean` | A flag to skip ignore cache and always fetch JWKs before each jwt verification. |

Reviewer: ```suggestion
| `skipJwksCache?` | `boolean` | A flag to ignore the JWKS cache and always fetch JWKS before each jwt verification. |
```

---
title: verifyToken()
description: Use Clerk's Backend SDK to
---

# `verifyToken()`

```js
function verifyToken: (token: string, options: VerifyTokenOptions) => Promise<JwtReturnType<JwtPayload, TokenVerificationError>>;
```

## Parameters

| Name | Type | Description |
| --- | --- | --- |
| `token` | `string` | The token to verify. |
| `options` | `VerifyTokenOptions` | Options for verifying the token. |

### `VerifyTokenOptions`

| Name | Type | Description |
| --- | --- | --- |
| `audience?` | `string \| string[]` | A string or list of [audiences](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3). |
| `authorizedParties`| `string[]` | | A list of authorized parties. For example, `['http://localhost:4003', 'https://clerk.com']` |
| `clockSkewInMs?` | `number` | Specifies the allowed time difference (in milliseconds) between the Clerk server (which generates the token) and the clock of the user's application server when validating a token. Defaults to 5000 ms (5 seconds).|

> | `jwtKey?` | `string` | The PEM public key from the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys) page -> Show JWT public key -> JWKS URL** section of the Clerk Dashboard. It's recommended to use [the environment variable](/docs/deployments/clerk-environment-variables) instead. |

Author: ```suggestion
| `jwtKey?` | `string` | The JWT Public Key from the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys) page -> Show JWT public key -> JWKS URL** section of the Clerk Dashboard. It's recommended to use [the environment variable](/docs/deployments/clerk-environment-variables) instead. |
```
